src/sidebar/components/tag-editor.js

```diff
@@ -1,35 +1,187 @@
+      const suggestionsSet = [];
+      suggestions.forEach(suggestion => {
+        if (tags.indexOf(suggestion) < 0) {
+          suggestionsSet.push(suggestion);
+        }
+      });
+      return suggestionsSet;
+    };
+    // Call filter with an empty string to return all suggestions
+    return removeDuplicates(tagsService.filter(''), tagList);
+  }, [tagsService, tagList]);
+
+  /**
+   * Handle changes to this annotation's tags
+   */
+  const updateTags = tagList => {
+    // update suggested tags list via service
+    tagsService.store(tagList.map(tag => ({ text: tag })));
+    // invoke callback with updated tags
+    onEditTags({ tags: tagList });
   };
 
-  this.autocomplete = function(query) {
-    return Promise.resolve(tags.filter(query));
+  /**
+   * Remove a tag from this annotation.
+   *
+   * @param {string} tag
+   */
+  const removeTag = tag => {
+    const newTagList = [...tagList]; // make a copy
+    const index = newTagList.indexOf(tag);
+    newTagList.splice(index, 1);
+    updateTags(newTagList);
   };
 
-  this.$onChanges = function(changes) {
-    if (changes.tags) {
-      this.tagList = changes.tags.currentValue.map(function(tag) {
-        return { text: tag };
-      });
+  /**
+   * Adds a tag to the annotation equal to the value of the input field
+   * and then clears out the suggestions list and the input field.
+   */
+  const addTag = () => {
+    const value = inputEl.current.value.trim();
+    if (value.length === 0) {
+      // don't add an empty tag
+      return;
+    }
+    if (tagList.indexOf(value) >= 0) {
+      // don't add duplicate tag
+      return;
+    }
+    updateTags([...tagList, value]);
+    setShowSuggestions(false);
+
+    // clear the input field and maintain focus
+    inputEl.current.value = '';
+    inputEl.current.focus();
+  };
+
+  /**
+   *  If the user pressed enter or comma while focused on
+   *  the <input>, then add a new tag.
+   */
+  const handleKeyPress = e => {
+    if (e.key === 'Enter' || e.key === ',') {
+      addTag();
+      // preventDefault stops the delimiter from being
+      // added to the input field
+      e.preventDefault();
     }
   };
+  /**
+   *  As the user types, change the suggestions query
+   *  based on the value of inputEl.
+   */
+  const handleKeyUp = e => {
+    if (e.key) {
+      // Show the suggestions if the user types anything
+      setShowSuggestions(true);
+    } else if (e.type === 'keyup') {
+      // Assume the user selected an option from <datalist>
+      // either by clicking on it with the mouse or pressing
+      // the 'Enter' key.
+      addTag();
+    }
+  };
+
+  const suggestionsList = () => {
+    return (
+      <datalist
+        id={`tag-editor-datalist-${annotation.id}`}
+        className="tag-editor__suggestions"
+        aria-label="Annotation suggestions"
+      >
+        {' '}
+        {suggestions.map((suggestion, index) => (
+          <option key={index} value={suggestion} />
```

Comment:
When I tested in Firefox I think I found that it didn't sort suggestions alphabetically, whereas Safari and Chrome did. If that is the case, it would be helpful if we sorted them before creating the options list.